xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-code-oldcore/src/main/java/org/xwiki/rendering/internal/macro/code/source/AbstractEntityCodeMacroSourceFactory.java

```diff
@@ -0,0 +1,140 @@
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceResolver;
+import org.xwiki.rendering.macro.MacroExecutionException;
+import org.xwiki.rendering.macro.code.source.CodeMacroSource;
+import org.xwiki.rendering.macro.code.source.CodeMacroSourceFactory;
+import org.xwiki.rendering.macro.code.source.CodeMacroSourceReference;
+import org.xwiki.rendering.transformation.MacroTransformationContext;
+import org.xwiki.security.authorization.AccessDeniedException;
+import org.xwiki.security.authorization.AuthorizationManager;
+import org.xwiki.security.authorization.Right;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Provide content coming from XWiki model entities.
+ * 
+ * @version $Id$
+ * @since 15.0RC1
+ * @since 14.10.2
+ */
+public abstract class AbstractEntityCodeMacroSourceFactory implements CodeMacroSourceFactory
+{
+    @Inject
+    @Named("macro")
+    private EntityReferenceResolver<String> resolver;
+
+    @Inject
+    private Provider<XWikiContext> xcontextProvider;
+
+    @Inject
+    private AuthorizationManager authorization;
+
+    @Inject
+    private ComponentManager componentManager;
+
+    @Override
+    public CodeMacroSource getContent(CodeMacroSourceReference reference, MacroTransformationContext context)
+        throws MacroExecutionException
+    {
+        // Resolve the reference
+        EntityReference entityReference =
+            this.resolver.resolve(reference.getReference(), getEntityType(), context.getCurrentMacroBlock());
+
+        XWikiContext xcontext = this.xcontextProvider.get();
+
+        if (xcontext == null) {
+            throw new MacroExecutionException("No XWiki context could be found in the current context");
+        }
+
+        // Resolve the document reference
+        DocumentReference documentReference = xcontext.getWiki().getDocumentReference(entityReference, xcontext);
+
+        // Current author must have view right on the document
+        try {
+            this.authorization.checkAccess(Right.VIEW, xcontext.getAuthorReference(), documentReference);
+        } catch (AccessDeniedException e) {
+            throw new MacroExecutionException(
+                "Current author is not allowed to access document [" + documentReference + "]");
+        }
+
+        // Current user must have view right on the document
+        if (this.authorization.hasAccess(Right.VIEW, xcontext.getUserReference(), documentReference)) {
+            throw new MacroExecutionException(
+                "Current user is not allowed to access document [" + documentReference + "]");
+        }
+
+        // Get the content
+        XWikiDocument document;
+        try {
+            document = xcontext.getWiki().getDocument(entityReference, xcontext);
```

Comment:
I doubt it changed anything but a bit cleaner to call it with `documentReference` since you performed the right checks with that one and you log that one in case of problem.

Reply:
Yes, it's better, I just missed this one when I modified the code.